core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/SSHActionTest.java

```diff
@@ -0,0 +1,102 @@
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.hydrator.plugin.batch.ETLBatchTestBase;
+import co.cask.hydrator.plugin.common.EchoCommandFactory;
+import co.cask.hydrator.plugin.common.MockActionContext;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.sshd.common.config.keys.KeyUtils;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.pubkey.PublickeyAuthenticator;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.server.session.ServerSession;
+import org.apache.sshd.server.shell.ProcessShellFactory;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.security.PublicKey;
+
+/**
+ * Test for {@link SSHAction}
+ */
+public class SSHActionTest extends ETLBatchTestBase {
+  // Test keypair config
+  private static final String PRIVATE_KEY =
```

Comment:
Another option is to programmatically generate a keypair for each test run but unsure if such a variable changing from run to run would be ideal.